### ISSUE TITLE
cumprod operation

### ISSUE THREAD

**Issue by claverru** [onnx/onnx#6590]
### System information

ONNX version: 1.17.0

### Are you willing to contribute it (Y/N)

None

### Notes

I just encountered while trying to serialize a torch model into ONNX that this operation is not yet supported. Like, is it such a strange operation?

I did a workaround by `x.log().cumsum().exp()` but it's hella slower.

Also how is it possible that cumsum is supported but cumprod isn't?

Thank you so much.

**Comment by justinchuby** [comment]
It is true that the op is not in the current ONNX spec. Could you share the use case? We can discuss adding a new definition

**Comment by claverru** [comment]
@justinchuby I don't think I understand the question.

I want to have the cumprod operation computed inside the ONNX model, as it fastest than any alternative and convenient for my concrete use case.

**Comment by justinchuby** [comment]
Yes, for example, what model is this op being used in? What type of layers etc. does it appear in?

**Comment by claverru** [comment]
How does that change the fact that I need that operation? I still don't understand. If you're thinking about a workaround, there isn't. I need to compute the cumulative product along an axis. And making log -> cumsum -> exp is slower than applying the cumprod after the model in numpy. 

Let me know if you plan on working on this or I should opt for alternatives.

There's cumprod operation in TensorFlow, Jax, Numpy and Torch. It surprised me it doesn't exist in ONNX. 

Thanks!

**Comment by justinchuby** [comment]
The operators in onnx tend to be what the community has been needing for their models, and cumprod hasn’t been one of them. It would be helpful we have a concrete example to motivate the inclusion of this op. If you or anyone would like to contribute, you may refer to https://github.com/onnx/onnx/blob/main/docs/AddNewOp.md for a guide. Thanks!

**Comment by claverru** [comment]
I want to compute the cumulative density function of survivability inside my model graph, given that I'm predicting N steps, being each step S the probability of survival given survivability until S.

This is the first time using ONNX. Wouldn't have expected not to have this operator in it.

Thanks mate.

**Comment by cbourjau** [comment]
+1 for adding `CumProd` to the standard. [`cumulative_prod`](https://data-apis.org/array-api/draft/API_specification/generated/array_api.cumulative_prod.html#array_api.cumulative_prod) is now also part of the array-api standard making this function generally more prominent and relevant to ndonnx, too. The function is easy to implement by runtimes. Not having such a function (or using the costly workaround mentioned in the OP behind the scenes) is a bad user experience.

**Comment by celmore25** [comment]
Here is a PR for this operator: https://github.com/onnx/onnx/pull/7075

**Comment by justinchuby** [comment]
This is added in https://onnx.ai/onnx/operators/onnx__CumProd.html
